src/client/js/pie-item.js

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ *  Online store PWA sample.
+ *  Copyright 2017 Google Inc. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ *
+ */
+ (function() {
+
+  const template = document.createElement('template');
+  template.innerHTML = `
+    <pie-img width="300px" height="200px"></pie-img>
+    <div>
+      <h1 class="pie-title skeleton skeleton-on"></h1>
+      <div class="pie-description skeleton-text skeleton-on">
+        <p></p>
+        <ul>
+          <li><span class="pie-weight"></span></li>
+          <li><span class="pie-price"></span></li>
+        </ul>
+      </div>
+    </div>`;
+
+  class PieItem extends HTMLElement {
```

Comment:
We should follow Rob Dodson's guidance on interoperable custom elements, namely using `attributeChangedCallback` to link attributes to properties. See http://robdodson.me/interoperable-custom-elements/ for the pattern to follow.

This is valuable even in a vanilla application like Pie Shop, because it helps us demonstrate best practices.

Reply:
I am using properties as source of truth here. I do not reflect to attributes, because I pass in an object and would need to serialize it - which is a pain and there is no need for it.
These elements are relevant only for this project, and there's no need to make them interoperable, so no point reflecting to attributes just for the sake of it, imo. Wdyt?

Reply:
That makes sense, no need to look at serialising objects. LGTM.